internal/exporter.go

```diff
@@ -52,6 +53,33 @@ type Exporter struct {
 	secretsCache *cache.Cache
 }
 
+type KubeSecretType struct {
+	Type   string
+	Regexp *regexp.Regexp
+}
+
+func ParseSecretType(s string) (KubeSecretType, error) {
+	ty, pattern, found := strings.Cut(s, ":")
+	if !found {
+		return KubeSecretType{}, errors.New("secret type needs to contain at least a single colon")
+	}
+	compiled, err := regexp.Compile(pattern)
+	if err != nil {
+		return KubeSecretType{}, err
+	}
+	return KubeSecretType{
+		Type:   ty,
+		Regexp: compiled,
+	}, nil
+}
+
+func (kst *KubeSecretType) Matches(secretType, key string) bool {
```

Comment:
I like the way you implemented this as a struct. Could it be possible to add the following function? The goal would be to use it in places where `Matches` is used inside a loop, it would make the code more readable by decreasing the number of nested loops. (warning: I did not tested neither compiled it).

```golang
func (kst *KubeSecretType) FindMatchingKey(secret *v1.Secret, secretType string, key string) string {
	for key := range secret.Data {
		if secretType.Matches(string(secret.Type), key) {
			return key
		}
	}

	return ""
}
```

Reply:
The function you outlined would only return the first key matching a regular expression. Some iterator with a callback could be handy though, e.g.:
```go
func IterateMatching(types []KubeSecretType, secret *v1.Secret, cb func(string)) {
	for _, secretType := range types {
		for key := range secret.Data {
			if secretType.Matches(string(secret.Type), key) {
				cb(key)
			}
		}
	}
}
```
That iteration could also deduplicate keys, when a key is matched by multiple regular expressions.